cobc/pplex.l

```diff
@@ -1323,6 +1330,67 @@ ppparse_error (const char *err_msg)
 	cb_plex_error (newline_count, "%s", err_msg);
 }
 
+
+void
+cobc_set_source_format (const enum cb_format sf) {
+	cb_source_format = sf;
+
+	switch (cb_source_format) {
+	case CB_FORMAT_VARIABLE:
+		/* This value matches most MF Visual COBOL 4.0 version. */
+		/* CHECKME: 500 seen in format_type entry of `ppparse.y`. */
+		cb_text_column = 250;
+		break;
+	default:
+		cb_text_column = cb_config_text_column;
+		break;
+	}
+
+	switch (cb_source_format) {
+	case CB_FORMAT_ACUTERM:
+	case CB_FORMAT_COBOLX:
+		cb_indicator_column = 1;
+		break;
+	default:
+		cb_indicator_column = 7;
+		break;
+	}
+}
+
+enum cb_format
+cobc_get_source_format (void) {
+	return cb_source_format;
+}
+
+static int
+cobc_parse_source_format (enum cb_format *const out, const char *const sfname) {
+	enum cb_format format;
+	if (!cb_strcasecmp (sfname, "FIXED")) {
+		format = CB_FORMAT_FIXED;
+	} else if (!cb_strcasecmp (sfname, "FREE")) {
+		format = CB_FORMAT_FREE;
+	} else if (!cb_strcasecmp (sfname, "VARIABLE")) {
+		format = CB_FORMAT_VARIABLE;
+	} else if (!cb_strcasecmp (sfname, "ACUTERM")) {
```

Comment:
Rechecked MF docs- after their full integration of ACUCOBOL-GT they actually added a
`SOURCEFORMAT"TERMINAL"` compiler directive which means exactly this, so lets use `TERMINAL` instead of `ACUTERM` for the text and stay with `CB_FORMAT_ACUTERM` for the enum value, despite my initial "this likely raises conflicts" thoughts - and handle the conflicts when/if they occur for a user.

Reply:
... ok the conflicts are already mentioned in the MF docs:

> ACUCOBOL-GT Terminal format is compatible with the VAX COBOL terminal source format, except for the introduction of the identification area, which VAX COBOL does not support. ACUCOBOL-GT Terminal mode is also compatible with ICOBOL terminal source format, except that area A must start in column 1 (in ICOBOL area A may start in column 2).

Any thoughts about possibly handling this? We should at least "somewhere" copy the facts of this text (not the text) into a comment in cobc, maybe `set_source_format()` is the best place.

Reply:
Does "area A may start in column 2" mean that margin B is "sometimes" shifted one character right in ICOBOL, like if the line starts with a space? The "may" is confusing. If it's a "must", it's COBOLX with `\D`, right?

Reply:
For ICOBOL I had a look at [their docs](https://www.icobol.com/ftproot/546/referic.pdf). Finding out there are three formats:

* "ANSI Card Format" -> -fformat-fixed
* "xCard (extended card)" format - which is identical to MF's VARIABLE format (but we should increase the right margin to 255, if possible)
* Free-Form Format (CRT) - _identical_ to acuterm but without ` \` before the `D`, so as with ACU there's a "floating" area B (always 4 characters from the indicator area and indicator area does only exist if there's an appropriate value in there)

Reply:
ICOBOL would be 255 while MF would be 250, and as we don't have `-fformat=variable` until "very soon" people actually used `-ftext-column=250` instead.

But we may consider dropping that option in 4.x (dropping `-fixed`/`-free` is likely too intrusive so those will stay even in 4.x).